website/client/src/components/payments/selectUserModal.vue

```diff
@@ -65,26 +67,17 @@
         name="selectUser"
         novalidate="novalidate"
       >
-        <div class="input-group">
-          <input
-            id="selectUser"
-            v-model="userSearchTerm"
-            class="form-control"
-            type="text"
-            ref="textBox"
-            :placeholder="$t('usernameOrUserId')"
-            :class="{
-              'input-valid': foundUser._id,
-              'is-invalid input-invalid': userNotFound,
-            }"
-          >
-        </div>
-        <div
-          v-if="userSearchTerm.length > 0 && userNotFound"
-          class="input-error text-center mt-2"
-        >
-          {{ $t('userWithUsernameOrUserIdNotFound') }}
-        </div>
+        <validated-text-input
+          id="selectUser"
+          v-model="userSearchTerm"
+          :is-valid="foundUser._id"
+
+          :placeholder="$t('usernameOrUserId')"
+          :invalid-issues="userSearchTerm.length > 0 && userNotFound
```

Comment:
This is quite awkward inline; perhaps this should be a function? It would be much easier to extend with further possible issues, that way.

Reply:
will move it out to a function